app.rb

```diff
@@ -0,0 +1,86 @@
+require './person'
+require './student'
+require './teacher'
+require './book'
+require './rental'
+require './classroom'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'Books: '
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+    puts ''
+  end
+
+  def list_people
+    puts 'People: '
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    puts ''
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)?'
+    print '[Input the number]: '
+    person_type = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp.to_s
+    print 'Age : '
+    age = gets.chomp.to_i
+    case person_type
+    when 1
+      print 'Has parent permission? [Y/N]: '
+      permission = gets[0]
+      permission = (permission == ('Y' || 'y'))
+      @people << Student.new(age, name, parent_permission: permission)
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(specialization, age, name)
+    end
```

Comment:
### Dear @r-ahmed2022 You have done an incredible job organizing the code. There are some issues we need to address though.
- [ ] After submitting the parent permission your app crashed and I could not continue to add a person. This error should be eliminated to have a better user experience and a working app that could achieve its purpose. 
- [ ] Also, because of this I could not continue to validate the rest of the functionalities. Please make all the functionalities are in place before submitting the next review request. 


https://user-images.githubusercontent.com/69077061/224336278-64f11a2b-1de2-400d-8a27-940ecfc83e9d.mov